src/main/java/org/truffleruby/core/rope/OnHeapRope.java

```diff
@@ -0,0 +1,20 @@
+package org.truffleruby.core.rope;
+
+import org.jcodings.Encoding;
+
+public abstract class OnHeapRope extends Rope {
```

Comment:
"OnHeap" as in opposition to Native?
How about MutableRope, is it a third kind of Rope?
Maybe Mutable/Immutable is a more useful distinction for caching?